### DISCUSSION TITLE
Export all Fruity Notebook text and all markers (with timestamps)?

### DISCUSSION THREAD

**Discussion by iFUCKINGHATEcomputers** [demberto/PyFLP#181] [Q&A]
I've been migrating all my notetaking and organization to Obsidian.md (it fucking rules), and there's a lot of stuff I have written inside my FLPs which I'd want to have exported in a single text file alongside each of them:

- any text written in Fruity Notebook 1/2
- text from time markers in each playlist/arrangement, timestamped 
- text from time markers in each pattern (with the names of which pattern/instrument they were written in)

Is this possible?

**Comment by demberto** [comment]
1. PyFLP supports only Fruity Notebook 2. The way its data is stored inside an FLP has changed across different versions of FL, so you might encounter a bug. Its tested to work on multiple pages atleast on the test FLP.
2. Timestamps aren't decoded into a human representation yet, so you will get a number. Consider timestamps to be not done yet. Timemarker names are supported.
3. Possible.
